vanetza/security/validity_restriction.hpp

```diff
@@ -35,6 +36,47 @@ class Duration
         return m_raw;
     }
 
+    /**
+     * Returns the duration unit.
+     */
+    Units unit() const
+    {
+        return static_cast<Units>(m_raw >> 13);
+    }
+
+    uint16_t value() const
+    {
+        return m_raw & 0xCFFFF; // mask upper 3 bit
+    }
+
+    /**
+     * Returns the duration in seconds.
+     */
+    Time64 to_seconds() const
```

Comment:
This is not possible: `Time64` encodes microseconds (see section 4.2.15 of TS 103 097).
Exactly because such conversion errors happen too easily with plain integers I would favor a solution based on `vanetza::Clock::duration`.